scene/main/viewport.cpp

```diff
@@ -2891,17 +2928,30 @@ int Viewport::gui_get_canvas_sort_index() {
 	return gui.canvas_sort_index++;
 }
 
-void Viewport::gui_release_focus() {
-	if (gui.key_focus) {
-		Control *f = gui.key_focus;
-		gui.key_focus = nullptr;
-		f->notification(Control::NOTIFICATION_FOCUS_EXIT, true);
-		f->queue_redraw();
+int Viewport::gui_get_active_focus_layer() const {
+	return gui.active_focus_layer;
+}
+void Viewport::gui_set_active_focus_layer(int p_focus_layer) {
+	gui.active_focus_layer = p_focus_layer;
+}
+
+void Viewport::gui_release_focus(int p_focus_layer) {
+	HashMap<int, Control *>::Iterator F = gui.focused_controls.find(p_focus_layer);
+	if (F) {
+		if (F->value) {
```

Comment:
Can `value` ever be null? The entry gets deleted when you release focus.

Reply:
Good point.